src/main/java/tuteez/model/person/lesson/Lesson.java

```diff
@@ -150,6 +152,58 @@ public static boolean isClashingWithOtherLesson(Lesson firstLs, Lesson secondLs)
         return !(completelyBefore || completelyAfter);
     }
 
+    /**
+     * Calculates the duration until the next occurrence of the lesson.
+     * If the lesson is currently ongoing, this method returns a zero duration.
+     *
+     * <p>The duration calculation considers both the day of the week and the time of day.
+     * If the current day matches the lesson day but the current time is after the lesson's end time,
+     * the method assumes the next lesson will occur in one week from the current day.</p>
+     *
+     * @return A {@code Duration} representing the time left until the next lesson,
+     *         or zero if the lesson is currently ongoing.
+     */
+    public Duration durationTillLesson() {
```

Comment:
Nice job implementing this function. I liked how you considered the case where the lesson is on the same day as the current day.